webhook_to_fedora_messaging/endpoints/service.py

```diff
@@ -0,0 +1,110 @@
+from flask import Blueprint, Flask, request, Response, Request
+from ..database import db
+from ..models.service import Service
+from ..models.user import User
+from sqlalchemy_helpers import get_or_create
+from .util import not_found, success, bad_request, created, conflict, validate_request, exclude_from_val
+
+app = Flask(__name__)
+service_endpoint = Blueprint("service_endpoint", __name__)
+
+
+
+@service_endpoint.route("/service", methods=["POST"])
+def create_service():
+    """
+        Used for creating a new service by sending a post request to /service/ path.
+    """
+    
+    session = db.Session()
+    body = request.json
+    
+    
+    user = session.query(User).filter(User.username == body['username']).first()
+    if user is None:
+        return not_found()
+    
+    service, is_created = get_or_create(session, Service, name=body['name'], type=body['type'], desc=body['desc'], user_id=user.id)
+    
+    if not is_created:
+        return conflict({'message': 'Service Already Exists'})
+    else:
+        return created({'message': 'Created', 'uuid': service.id})
+        
+    
+@service_endpoint.route("/service/search", methods=["GET"])
+def list_services():
+    session = db.Session()
+    user = session.query(User).filter(User.username.like(request.json['username'])).first()
+    if user is None:
+        return not_found()
+    
+    services = session.query(Service).filter(Service.user_id == user.id).all()
+    
+    return success({'service_list': services})
+    
+    
+@service_endpoint.route("/service", methods=["GET"])
+@exclude_from_val
+def lookup_service():
+    if not validate_request(request, ['service_uuid']):
+        return bad_request()
+    
+    session = db.Session()
+    service = session.query(Service).filter(Service.id == request.json['service_uuid']).first()
+    
+    if service is None:
+        return not_found()
+    else:
+        return success({'uuid': service.id, 'name': service.name, 'type': service.type, 'desc': service.desc})
+    
+    
+#TODO make the api key also disabled or what needed here.
+@service_endpoint.route("/service/revoke", methods=["PUT"])
+@exclude_from_val
+def revoke_service():
+    
+    if not validate_request(request, fields=['username', 'service_uuid']):
+        return bad_request()
+    
+    session = db.Session()
+    user = session.query(User).filter(User.username == request.json['username']).first()
+    if user is None:
+        return not_found()
+    
+    service = session.query(Service).filter(Service.user_id == user.id).filter(Service.id == request.json['service_uuid']).first()
+    if service is None:
+        return not_found()
+    
+    service.disabled = True
+    session.commit()
+    
+    
+    return success({'uuid': service.id, 'is_valid': not service.disabled})
+    
+    
+
+@service_endpoint.route("/service", methods=["PUT"])
+@exclude_from_val
+def update_service():
```

Comment:
If docstrings are created for a function of a kind, please ensure that they are created for other functions of that kind.